java/src/jmri/util/StringUtil.java

```diff
@@ -533,6 +533,28 @@ static public String arrayToString(@Nonnull int[] v) {
         }
         return new String(retval);
     }
+    
+    /**
+     * Convert an array of integers into a single hex. string. 
+     * Each element is displayed consecutively, no spaces.
+     * <p>
+     * eg. int[]{1,2,3,10} will return String "0102030A"
+     * eg. int[]{-1} will return "FFFFFFFF"
+     * eg. int[]{256} will return "100"
+     * eg. int[]{257} will return "101"
+     *
+     * @param v the array of integers
+     * @return the formatted String or an empty String
+     */
+    @CheckReturnValue
+    @Nonnull
+    static public String arrayToHexString(@Nonnull int[] v) {
```

Comment:
Question: what problem is this solving that hexStringFromBytes doesn’t?

Reply:
Thanks for taking the time to look over, good sanity check, will remove.

Reply:
Hi @pabender , the main issue I'm having while attempting to use hexStringFromBytes is the int[] to byte[] conversion.
The ByteBuffer / IntBuffer route outputs a 4 character string when we need a 2 character, all of the other methods I've seen involve a loop.  If we're looping, we may as well extract the String value there and then.

To make things a bit clearer, have renamed the new method to hexStringFromInts and improved the javadoc,

Steve.